### PR TITLE
refactor: add zIndex object to base theme - FE-2765

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
Adding zIndex object will help us to unify our zindex components and also it will avoid overlap components 
```js
  //base-theme.config.js
  zIndex: {
      overlay: 1000,
      popover: 2000,
      modal: 3000,
      header: 4000,
      fullScreenModal: 5000,
      notification: 6000
    }

// component.style.js
  zIndex: ${({theme}) => theme.zIndex.popover}

```

### Current behaviour
```js
// component.style.js
  zIndex: 2000;
```

### Checklist

- [ ] <del> Carbon implementation and Design System documentation are congruent
- [x] Release notes (Conventional Commits) 
- [x] Unit tests
- [x] Cypress automation tests
- [x] Storybook added or updated
- [ ] <del> Theme support
- [ ] <del> Typescript `d.ts` file added or updated

### CODE REVIEW

File: src/components/action-popover/__snapshots__/action-popover.spec.js.snap [hunk 1]
```diff
@@ -1283,7 +1285,9 @@ exports[`ActionPopover submenu left aligned closes the submenu when right key is
   background-color: ",
               [Function],
               ";
-  z-index: 1;
+  z-index: ",
+              [Function],
+              ";
```
Comment (mkrds) [inline src/components/action-popover/__snapshots__/action-popover.spec.js.snap:1290]: That looks weird

File: src/components/action-popover/__snapshots__/action-popover.spec.js.snap [hunk 2]
```diff
@@ -2609,7 +2615,9 @@ exports[`ActionPopover submenu left aligned opens the submenu on mouseenter 1`]
...
```
Comment (mkrds) [inline src/components/action-popover/__snapshots__/action-popover.spec.js.snap:2620]: Same here

File: src/components/action-popover/__snapshots__/action-popover.spec.js.snap (hunk 1 above)
Comment (dawidzarzycki) [reply src/components/action-popover/__snapshots__/action-popover.spec.js.snap:1290]: Good point! I will investigate this

**Pull request merged**